src/robotide/ui/tagdialogs.py

```diff
@@ -100,6 +101,16 @@ def _make_bindings(self):
         self._tags_list.Bind(wx.EVT_LIST_ITEM_RIGHT_CLICK, self.OnRightClick)
         self._tags_list.Bind(wx.EVT_LIST_COL_CLICK, self.OnColClick)
 
+    def _tag_name_for_sort(self, tag_name):
+        mylist = [part if index % 2 == 0 else int(part) for index, part in
+                  enumerate(re.split(r'(\d+)', tag_name.lower()))]
+        # Side effect from using regex, u'' will be removed from start or end
+        if mylist[0] == unicode(''):
+            del mylist[0]
+        if mylist[-1] == unicode(''):
+            del mylist[-1]
```

Comment:
Empty strings shouldn't be removed. Keeping them enforces that the returned sort key always starts and ends with a string and is thus comparable. Sorting, for example, `('foo', 1)` and `(2, 'bar')` makes no sense (in Python 3 it even raises `TypeError`), but comparing `('foo', 1, '')` and `('', 2, 'bar')` is fine.